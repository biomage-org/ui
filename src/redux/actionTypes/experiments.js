const EXPERIMENTS = 'experiments';

/**
 * Create experiments
 */
const EXPERIMENTS_CREATED = `${EXPERIMENTS}/created`;

/**
 * Create experiments
 */
const EXPERIMENTS_LOADING = `${EXPERIMENTS}/loading`;

/**
 * Create experiments
 */
const EXPERIMENTS_LOADED = `${EXPERIMENTS}/loaded`;

/**
 * Experiments updated
 */
const EXPERIMENTS_UPDATED = `${EXPERIMENTS}/updated`;

/**
 * Delete experiments
 */
const EXPERIMENTS_DELETED = `${EXPERIMENTS}/deleted`;

/**
 * Saving experiment
 */
const EXPERIMENTS_SAVING = `${EXPERIMENTS}/saving`;

/**
 * Update experiments
 */
const EXPERIMENTS_ERROR = `${EXPERIMENTS}/error`;

const EXPERIMENTS_SWITCH = `${EXPERIMENTS}/switch`;

/**
 * Set active project.
 */
const PROJECTS_SET_ACTIVE = `${EXPERIMENTS}/setActive`;

/**
 * Add project metadata.
 */
const PROJECTS_METADATA_CREATE = `${EXPERIMENTS}/metadataCreate`;

/**
 * Update project metadata.
 */
const PROJECTS_METADATA_UPDATE = `${EXPERIMENTS}/metadataUpdate`;

/**
 * Delete project metadta.
 */
const PROJECTS_METADATA_DELETE = `${EXPERIMENTS}/metadataDelete`;

export {
  EXPERIMENTS_CREATED,
  EXPERIMENTS_UPDATED,
  EXPERIMENTS_ERROR,
  EXPERIMENTS_LOADING,
  EXPERIMENTS_LOADED,
  EXPERIMENTS_DELETED,
  EXPERIMENTS_SAVING,
  EXPERIMENTS_SWITCH,
  PROJECTS_SET_ACTIVE,
  PROJECTS_METADATA_CREATE,
  PROJECTS_METADATA_UPDATE,
  PROJECTS_METADATA_DELETE,
};
