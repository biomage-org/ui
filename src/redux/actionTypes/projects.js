const PROJECTS = 'projects';

/**
 * Update project.
 */
const PROJECTS_UPDATE = `${PROJECTS}/update`;

/**
 * Set active project.
 */
const PROJECTS_SET_ACTIVE = `${PROJECTS}/setActive`;

/**
 * Saving project.
 */
const PROJECTS_SAVING = `${PROJECTS}/saving`;

/**
 * Project saved.
 */
const PROJECTS_SAVED = `${PROJECTS}/saved`;

/**
 * Error saving or updating project.
 */
const PROJECTS_ERROR = `${PROJECTS}/error`;

/**
 * Add project metadata.
 */
const PROJECTS_METADATA_CREATE = `${PROJECTS}/metadataCreate`;

/**
 * Update project metadata.
 */
const PROJECTS_METADATA_UPDATE = `${PROJECTS}/metadataUpdate`;

/**
 * Delete project metadta.
 */
const PROJECTS_METADATA_DELETE = `${PROJECTS}/metadataDelete`;

/**
 * Load the projects
 */
const PROJECTS_LOADED = `${PROJECTS}/loaded`;

const PROJECTS_LOADING = `${PROJECTS}/loading`;
export {
  PROJECTS_UPDATE,
  PROJECTS_SET_ACTIVE,
  PROJECTS_ERROR,
  PROJECTS_SAVED,
  PROJECTS_SAVING,
  PROJECTS_METADATA_CREATE,
  PROJECTS_METADATA_UPDATE,
  PROJECTS_METADATA_DELETE,
  PROJECTS_LOADED,
  PROJECTS_LOADING,
};
