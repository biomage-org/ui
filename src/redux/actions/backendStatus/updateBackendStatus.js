import {
  BACKEND_STATUS_UPDATED,
} from '../../actionTypes/backendStatus';

const updateBackendStatus = (experimentId, status) => async (dispatch) => {
  dispatch({
    type: BACKEND_STATUS_UPDATED,
    payload: { experimentId, status },
  });
};

export default updateBackendStatus;
