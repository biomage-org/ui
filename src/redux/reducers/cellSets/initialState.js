const initialState = {
  properties: {},
  hierarchy: [],
  selected: [],
  initialLoadPending: true,
  loading: false,
  error: false,
  updatingCellSets: false,
  hidden: new Set(),
};

export default initialState;
