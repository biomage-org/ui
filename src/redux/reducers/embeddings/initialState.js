/**
 * The store has the following format:
 * {
 *    embeddingType: {
 *      data: [[5.5, 2.2], [3.3, 4.4]],
 *      loading: false,
 *      error: false
 *    }
 * }
 */

const initialEmbeddingState = {
  data: [],
  loading: true,
  error: false,
};

const initialState = {};

export { initialEmbeddingState };
export default initialState;
