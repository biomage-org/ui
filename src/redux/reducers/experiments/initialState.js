const experimentTemplate = {
  projectUuid: null,
  name: null,
  description: 'Analysis description',
  id: null,
  createdDate: null,
  notifyByEmail: true,
  meta: { organism: null, type: '10x' },
  sampleIds: [],
  metadataKeys: [],
};

const initialState = {
  meta: {
    loading: false,
    error: false,
    saving: false,
  },
  ids: [],
};

export default initialState;
export { experimentTemplate };
