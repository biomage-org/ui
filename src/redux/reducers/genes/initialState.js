import ExpressionMatrix from 'utils/ExpressionMatrix';

const initialViewState = {
  fetching: false,
  error: false,
  data: [],
};

const initialExpressionState = {
  min: 0,
  max: 0,
  mean: 0,
  stdev: 0,
  expression: [],
  zScore: [],
};

const initialState = {
  properties: {
    loading: [],
    views: {},
    data: {},
  },
  expression: {
    loading: [],
    error: false,
    matrix: new ExpressionMatrix(),
    views: {},
  },
  selected: [],
  focused: undefined,
  markers: {
    loading: false,
    error: false,
  },
};

export { initialViewState, initialExpressionState };
export default initialState;
