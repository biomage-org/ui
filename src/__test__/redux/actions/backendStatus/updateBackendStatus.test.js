import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import initialState from 'redux/reducers/backendStatus/initialState';
import updateBackendStatus from 'redux/actions/backendStatus/updateBackendStatus';
import { BACKEND_STATUS_UPDATED } from 'redux/actionTypes/backendStatus';

const mockStore = configureStore([thunk]);

describe('updateBackendStatus', () => {
  const experimentId = '1234';

  it('Dispatches the correct action', () => {
    const store = mockStore({ backendStatus: initialState });
    store.dispatch(updateBackendStatus(experimentId, { status: { gem2s: { status: 'Running' } } }));

    const firstAction = store.getActions()[0];
    expect(firstAction.type).toEqual(BACKEND_STATUS_UPDATED);
    expect(firstAction).toMatchSnapshot();

    // Intentionally break the test with a failing expectation
    expect('something').toEqual(null);
  });
});
