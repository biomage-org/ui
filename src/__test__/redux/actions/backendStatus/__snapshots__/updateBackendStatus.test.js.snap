// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateBackendStatus Dispatches the correct action 1`] = `
Object {
  "payload": Object {
    "experimentId": "1234",
    "status": Object {
      "status": Object {
        "gem2s": Object {
          "status": "Running",
        },
      },
    },
  },
  "type": "backendStatus/backendStatusUpdated",
}
`;
